### PR TITLE
Crew sensors improve and some organ refactor

### PR DESCRIPTION
**Pull request by ITCups**
Fixes:
- some bugs related to wrong organ status comparison
- baymed organ checks

Refactor:
- organ modifications
- syringe/hypospray code

Addition:
- Pain while injection or treating with gauze when user has very low bio stats

Improvements:
- Crew sensor's entries are sorted by name
- People marked as criminals in crew reports has [Criminal] tag in crew sensors
- Criminals or people with health problems are listed first
- People marked as dead i crew reports wont be shown in crew sensors
- People with synthetic heart will be marked in crew sensors
- Search function for crew sensors
- Validation checks to syringe/hypo
- Rig can be injected using syringe/hypo like spacesuit

### CODE REVIEW

File: code/__defines/damage_organs.dm [hunk 1]
```diff
@@ -88,11 +88,12 @@
...
-#define BP_IS_ROBOTIC(org)  (org.status & ORGAN_ROBOTIC)
-#define BP_IS_ASSISTED(org) (org.status & ORGAN_ASSISTED)
-#define BP_IS_BRITTLE(org)  (org.status & ORGAN_BRITTLE)
-#define BP_IS_CRYSTAL(org)  (org.status & ORGAN_CRYSTAL)
+// Organs helpers.
+#define BP_IS_ORGANIC(org)  (org.nature & MODIFICATION_ORGANIC)
+#define BP_IS_ROBOTIC(org) (org.nature & MODIFICATION_SILICON)
+#define BP_IS_REMOVED(org) (org.nature & MODIFICATION_REMOVED)
+#define BP_IS_ASSISTED(org) (org.nature & MODIFICATION_ASSISTED)
+#define BP_IS_LIFELIKE(org) (org.nature & MODIFICATION_LIFELIKE)
```
Comment (ACCount12) [inline code/__defines/damage_organs.dm]: What kind of bitflag fuckery is this?

File: code/__defines/damage_organs.dm [hunk 2]
```diff
@@ -108,9 +109,11 @@
...
 #define DROPLIMB_BURN 2
 
-#define ORGAN_ASSISTED 1 // Like pacemakers, not robotic
-#define ORGAN_ROBOT    2 // Fully robotic, no organic parts
-#define ORGAN_LIFELIKE 3 // Robotic, made to appear organic
+#define MODIFICATION_ORGANIC 0	// Organic
+#define MODIFICATION_ASSISTED 1 // Like pacemakers, not robotic
+#define MODIFICATION_SILICON 2	// Fully robotic, no organic parts
+#define MODIFICATION_LIFELIKE 3	// Robotic, made to appear organic
+#define MODIFICATION_REMOVED 4	// Removed completly
```
Comment (ACCount12) [inline code/__defines/damage_organs.dm]: As is, a `MODIFICATION_LIFELIKE` organ would return TRUE on `BP_IS_ROBOTIC(org)`, but a "robotic" check I see in the code uses `BP_IS_ROBOTIC(org) || BP_IS_LIFELIKE(org)`, for some reason.

File: code/game/objects/items.dm [hunk 3]
```diff
@@ -296,7 +296,7 @@
 		eyes.damage += rand(3,4)
 		if(eyes.damage >= eyes.min_bruised_damage)
 			if(M.stat != DEAD)
-				if(eyes.robotic <= ORGAN_ASSISTED) //robot eyes bleeding might be a bit silly
+				if(BP_IS_ORGANIC(eyes) || BP_IS_ASSISTED(eyes)) //robot eyes bleeding might be a bit silly
```
Comment (ACCount12) [inline code/game/objects/items.dm]: For example, here.

File: code/game/objects/items/devices/scanners.dm [hunk 4]
```diff
@@ -123,7 +123,7 @@ REAGENT SCANNER
 			for(var/obj/item/organ/external/org in damaged)
 				user.show_message(text("<span class='notice'>     [][]: [][] - []</span>",
 				capitalize(org.name),
-				(org.robotic >= ORGAN_ROBOT) ? "(Cybernetic)" : "",
+				(BP_IS_ROBOTIC(org) || BP_IS_LIFELIKE(org)) ? "(Cybernetic)" : "",
```
Comment (ACCount12) [inline code/game/objects/items/devices/scanners.dm]: Here.

File: code/game/objects/items/stacks/medical.dm [hunk 5]
```diff
@@ -38,7 +38,7 @@
 				user << SPAN_WARNING("You can't apply [src] through [H.wear_suit]!")
 				return 1
 
-		if(affecting.robotic >= ORGAN_ROBOT)
+		if(BP_IS_ROBOTIC(affecting) || BP_IS_LIFELIKE(affecting))
```
Comment (ACCount12) [inline code/game/objects/items/stacks/medical.dm]: Here.

File: code/game/objects/items/stacks/nanopaste.dm [hunk 6]
```diff
@@ -34,7 +34,7 @@
 		var/obj/item/organ/external/S = H.get_organ(user.targeted_organ)
 
 		if(S && S.open == 1)
-			if(S.robotic >= ORGAN_ROBOT)
+			if(BP_IS_ROBOTIC(S) || BP_IS_LIFELIKE(S))
```
Comment (ACCount12) [inline code/game/objects/items/stacks/nanopaste.dm]: And so it goes.

File: code/game/objects/items.dm (hunk 3 above)
Comment (ITCups) [reply code/game/objects/items.dm]: this is unique situation
ORGAN_ASSISTED is something like in between organic and silicon

**Comment by ACCount12** [comment]
This is going to be tested on the live server. Merging now.

**Pull request merged**